### PR TITLE
test(web): add useIsMobile check harness

### LINKED ISSUE #107: Add a check harness for useIsMobile.ts
## What

`web/src/lib/useIsMobile.ts` has zero test coverage. It's used to branch layout on nearly every screen in the app (`Legal.tsx`, `BrandKit.tsx`, `Faq.tsx`, `Compare.tsx`, `ReceiveEntry.tsx`, `NotFound.tsx`, `TransferFlow.tsx`, `NearbyDevices.tsx`, and more) — it's the single hook the entire mobile-first design commitment in CLAUDE.md rests on.

## Why

The hook has real logic worth locking down: a synchronous initial read from `matchMedia` (avoiding a desktop→mobile layout flash on first paint), a guard for environments where `matchMedia` is undefined, and a `change`-event subscription with cleanup. A regression here — wrong breakpoint math, a missed listener cleanup — would silently affect mobile rendering across the entire site, and nothing today would catch it before a manual device-width check.

## Where to start

`web/src/lib/useIsMobile.ts` (whole file). Pattern-match against any existing engine check harness for the dependency-free `node` + stub style, e.g. `web/src/lib/warp/idbStage.check.mjs`.

## How to fix

Add `web/src/lib/useIsMobile.check.mjs` (or a small React Testing Library test if a test runner gets introduced for hooks like this one — check whether the project prefers to stay dependency-free per its existing convention). Cover: initial state matches the current viewport width, the breakpoint boundary itself, and that the `change` listener fires and cleans up on unmount.

## Acceptance criteria

- [ ] A runnable test/check exists covering the breakpoint logic, the SSR/undefined-`matchMedia` guard, and listener cleanup.
- [ ] `pnpm lint && pnpm typecheck && pnpm --filter @warp/web build` pass.

## Constraints

Keep it dependency-free if possible, matching the project's existing `.check.mjs` convention rather than introducing a new test framework just for this hook.

## Good first issue?

Yes — one small, well-scoped hook with clear, enumerable behaviors to lock down.

### PR DESCRIPTION
**Pull request by Pepsukka1**
Closes #107

Adds a dependency-free check harness for `useIsMobile.ts`, following Warp's existing `.check.mjs` convention.

Covers the synchronous initial media-query read, breakpoint behavior, change-event updates, the missing-`matchMedia` guard, and listener cleanup.

No production code changes. Tests/build were not run locally in this environment; CI can validate the branch.

## Summary by CodeRabbit

* **Tests**
  * Added automated coverage for mobile viewport detection across server-rendered and browser environments.
  * Verified default and custom breakpoints, including boundary values.
  * Added checks for initial responsive state, viewport change updates, event listener handling, and cleanup.
  * The validation runs without requiring additional runtime dependencies and skips gracefully when the test tooling is unavailable.

<details><summary><h3>Greptile Summary</h3></summary>

The PR adds a dependency-free check harness for `useIsMobile` and registers it with the standard CI check runner.
- Covers SSR and browser environments, including an existing `window` without `matchMedia`.
- Checks initial state, breakpoint boundaries, change events, and listener cleanup.
- Adds a discoverable wrapper under `src/lib/warp` so CI executes the colocated harness.
</details>

<h3>Confidence Score: 5/5</h3>

The PR appears safe to merge.

No blocking failure remains; the current code addresses all three previously reported issues.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| web/src/lib/useIsMobile.check.mjs | Adds the hook harness and resolves the prior error-masking and missing-`matchMedia` coverage findings. |
| web/src/lib/warp/useIsMobile.check.mjs | Registers the harness in the directory discovered by the standard CI runner. |

<sub>Reviews (2): Last reviewed commit: ["test: register useIsMobile harness in CI"](https://github.com/ishannaik/warp/commit/9e283762cc10930154c28d07408f16b24af631b8) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=51541494)</sub>

### CODE REVIEW

File: web/src/lib/useIsMobile.check.mjs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        },
+      },
+    ],
+  });
+  const code = out.outputFiles[0].text;
+  const dataUrl = `data:text/javascript;base64,${Buffer.from(code).toString("base64")}`;
+  ({ useIsMobile } = await import(dataUrl));
+} catch (error) {
+  console.error("SKIP: esbuild not available to transpile useIsMobile.ts —", error.message);
+  process.exit(0);
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply web/src/lib/useIsMobile.check.mjs]: Addressed by commit bad785f (fail closed): the harness no longer catches bundle or import errors, so regressions exit non-zero.

File: web/src/lib/useIsMobile.check.mjs [hunk 2]
```diff
@@ -0,0 +1,135 @@
+/**
+ * Dependency-free check for useIsMobile.ts. React's two hooks are stubbed so
+ * the hook can be driven from Node without adding a browser test framework.
+ *
+ * Run from web/: node src/lib/useIsMobile.check.mjs
+ */
+
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply web/src/lib/useIsMobile.check.mjs:7]: Addressed by commit 9e28376: the wrapper web/src/lib/warp/useIsMobile.check.mjs registers the harness with the CI runner, confirmed by run-checks.mjs discovery.

File: web/src/lib/useIsMobile.check.mjs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+import assert from "node:assert";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+
+const here = path.dirname(fileURLToPath(import.meta.url));
+
+let useIsMobile;
+try {
+  const esbuild = await import("esbuild");
+  const out = await esbuild.build({
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply web/src/lib/useIsMobile.check.mjs]: Addressed by commit bad785f: no catch block remains, so esbuild and load failures propagate as failures.

File: web/src/lib/useIsMobile.check.mjs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      for (const listener of listeners) listener({ matches: next });
+    },
+    listenerCount() {
+      return listeners.size;
+    },
+  };
+}
+
+// SSR / environments without matchMedia: safe desktop default.
+setWindow(undefined);
```
[the remark this answers is not in the document]
Comment (Ishannaik) [reply web/src/lib/useIsMobile.check.mjs]: The current harness tests the missing-matchMedia guard via setWindow({}) at line 66, distinct from the SSR no-window case.

**Comment by Ishannaik** [comment]
Verdict: MERGE

Tests: web check runner passes (all 17 engine harnesses, including the new useIsMobile check), typecheck passes, build passes.

Flags: no new network calls, no new dependencies (esbuild is already in web/package.json), no eval, no pickle, no workflow or release edits. The base64 data URL is a test-only import of the bundled hook, not shipped code.

The harness covers SSR, missing matchMedia, breakpoint boundaries, change events, and cleanup. All four review comments (fail-closed, CI registration, error masking, matchMedia case) were addressed in commits bad785f and 9e28376 and verified locally.
